Add basic release notes description (#1086)
kubernetes/sig-release: release-engineering/release-notes.md

# Kubernetes Release Notes Overview

The process around the creation of the Release Notes in Kubernetes falls into
the responsibility of SIG Release. Multiple sub-teams are involved in the
process as well as different publishing targets (or consumers). The current
state of the involved participants and tooling looks like this:

![release notes overview](img/kubernetes-release-notes.svg "Overview about the Kubernetes Release Notes")

During the release cycle, the **Release Notes Team** of SIG Release continuously
creates drafted Markdown and JSON versions of the ongoing release. The release

> notes website as well as this repository are getting updated during the whole

Author: Added :+1:

cycle.

The **Branch Managers** of the Release Engineering sub-team are updating the
main [Kubernetes repository](https://git.k8s.io/kubernetes) in parallel to the
ongoing process. This happens usually during the actual release cut in an
automated fashion.

> The **Documentation** sub-team takes care that the official

Reviewer: ```suggestion
The **Documentation** sub-team of the Release Team ensures that the official
```

> [Kubernetes.io](https://kubernetes.io) website gets updated after the final

Reviewer: ```suggestion
[kubernetes.io](https://kubernetes.io) website gets updated after the final
```

release version has been cut.

> Patch releases can happen every time during the release and will follow its own

Reviewer: ```suggestion
Patch releases happen on their own [cadence](/releases/patch-releases.md), separate from the active release
```

> schedule. The **Patch Release Team** of the Release Engineering sub-team takes

Reviewer: ```suggestion
schedule. The **Patch Release Team** (part of the [Release Managers group](/release-managers.md)) takes
```

> care of updating the repository and the website by their own.

Reviewer: ```suggestion
care of updating the relevant repositories during a patch release.
```
